pico/tools/src/main/java/io/helidon/pico/tools/CustomAnnotationTemplateRequest.java

```diff
@@ -98,4 +101,10 @@ public interface CustomAnnotationTemplateRequest {
      */
     GenericTemplateCreator genericTemplateCreator();
 
+    /**
+     * Current element that is processed, if available.
+     *
+     * @return processed element
+     */
+    Optional<Element> processedElement();
```

Comment:
Do we really want to leak this information out?  I think this sets a dangerous precedent for other cases.  Why did you feel you needed to do this anyway?

Reply:
Removed